src/compas_viewer/components/slider.py

```diff
@@ -65,58 +70,59 @@ def __init__(
         >>> slider = Slider("Brightness", min_val=0, max_val=255, starting_val=100)
         """
         super().__init__()
-        # TODO(pitsai): add step method
         self.title = title
         self.action = action
         self._horizontal = horizontal
         self.min_val = min_val
         self.max_val = max_val
-        self.starting_val = starting_val
+        self.step = step or 1
+        self.starting_val = starting_val if starting_val is not None else self.min_val
         self._tick_interval = tick_interval
 
-        if self._horizontal:
-            orientation = Qt.Horizontal
+        orientation = Qt.Horizontal if horizontal else Qt.Vertical
+
+        if self.step:
+            self.min_val /= step
+            self.max_val /= step
```

Comment:
I will not mutate the `min_val` and `max_val` like this, because a user might want to access the these two attributes later like what happens in  `scripts/sidedock.py` Line 21~22.  This creates unexpected behaviors.

Instead, you can create a read-only computed property for the scaled min and max value that are used internally:
```python
@property
def _scaled_min_val(self):
    return self.min_val / self.step

@property
def _scaled_max_val(self):
    return self.max_val / self.step
```

Reply:
good point!